### PR TITLE
UI: Fix can't paste removed item's filter issue

### PR DESCRIPTION
**Pull request by offthegrid-mike**
### Description

Previously: Paste filters is clickable after removing source after copy, but doesn't do anything.
Now changed to disable paste filter action if the item is removed

### Motivation and Context

Refer to issue #4075.
It is due to missing out checking the source item is still existed when setting up the context menu

### How Has This Been Tested?

1. Add any source.
2. Add any filters to that source.
3. Add another source (that can have these filters, e.g. both sources can have video filters)
4. Copy filters from the first source.
5. Remove the first source.
6. Right click -> Paste Filters on the second source.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by RytoEX** [comment]
As I mentioned in your other recent PR, please use an imperative verb in sentence case for the first word after the commit prefix ("Fix" instead of "fixed").

File: UI/window-basic-main.cpp [hunk 1]
```diff
@@ -3316,7 +3316,6 @@ void OBSBasic::VolControlContextMenu()
 				     QVariant::fromValue<VolControl *>(vol));
 
 	/* ------------------- */
-
```
Comment (RytoEX) [inline UI/window-basic-main.cpp:3319]: Don't make unnecessary whitespace changes.

**Review by RytoEX — CHANGES_REQUESTED** [review]
Don't change whitespace in unrelated areas of code.

File: UI/window-basic-main.cpp [hunk 2]
```diff
@@ -4553,7 +4553,13 @@ void OBSBasic::on_scenes_customContextMenuRequested(const QPoint &pos)
 	if (item) {
 		QAction *pasteFilters =
 			new QAction(QTStr("Paste.Filters"), this);
-		pasteFilters->setEnabled(copyFiltersString);
+		OBSSource filterSource = NULL;
+		if (copyFiltersString) {
+			filterSource =
+				obs_get_source_by_name(copyFiltersString);
+			obs_source_release(filterSource);
```
Comment (tt2468) [inline UI/window-basic-main.cpp:4560]: We recently added the OBSSourceAutoRelease wrapper, which avoid the need for the extra `obs_source_release()` call. As such, the wrapper should be used here.

**Comment by RytoEX** [comment]
Closing this as the associated Issue that it was meant to fix has been fixed by refactoring the copy/paste systems for sources and filters to use weak sources instead of the strings of source/filter names.

**Pull request closed without merging**